docs/cli_demo/bitcoind.md

```diff
@@ -0,0 +1,284 @@
+# Working with `Bitcoind`
+
+In this tutorial, we will guide you through setting up a Bitcoin Core `bitcoind` node in `regtest` mode. We'll cover basic operations like creating wallets, generating blocks, checking balances, and sending Bitcoin between two wallets.
+
+---
+
+### **1. Installing `bitcoind`**
+
+To start working with Bitcoin, it's necessary for us to install the Bitcoin Core software, which includes the `bitcoind` daemon.
+
+#### **Steps:**
+
+1. **Get Bitcoin Core:**
+
+   - We will visit the [official Bitcoin Core website](https://bitcoin.org/en/download) and download the appropriate version for our operating system.
+   - After downloading, we’ll follow the instructions on the site to verify the downloaded files by checking the signatures to ensure authenticity.
+
+2. **Verify Installation:**
+   - After installation, we can run the following command to confirm that `bitcoind` is properly installed:
+     ```bash
+     bitcoind --version
+     ```
+   - This will output the version of `bitcoind` if it's installed correctly.
+
+---
+
+### **2. Setting up a `bitcoin.conf` File**
+
+While `bitcoind` can be run on various Bitcoin networks, we will focus on the `regtest` network in this tutorial, which is a local blockchain environment ideal for development and testing. Before running `bitcoind`, we need to configure the node with a `bitcoin.conf` file to set up the `regtest` network.
+
+#### **Sample `bitcoin.conf` File for `regtest`**
+
+First, we’ll create a directory for Bitcoin data and configuration files if it doesn’t already exist:
+
+```bash
+mkdir -p ~/.bitcoin
+```
+
+Then, we’ll create a `bitcoin.conf` file in `~/.bitcoin/` and add the following lines:
+
+```ini
+regtest=1
+server=1
+fallbackfee=0.0001
+rpcuser=admin
+rpcpassword=password
+rpcbind=0.0.0.0:18443
+rpcallowip=0.0.0.0/0
+txindex=1
+```
+
+#### **Explanation of Configurations:**
+
+- `regtest=1`: Runs the node in regtest mode.
+- `server=1`: Enables `bitcoind` to run as a server and accept RPC (Remote Procedure Call) commands.
+- `rpcuser` and `rpcpassword`: Set the username and password for `bitcoin-cli` RPC access. We can customize these values or leave them as provided.
+- `rpcbind=0.0.0.0:18443`: Binds the RPC server to this IP and port.
+- `rpcallowip=0.0.0.0/0`: Allows RPC connections from any IP address. We should be cautious when using this in a non-development environment.
+- `fallbackfee=0.0001`: Specifies a fallback transaction fee when none is provided.
+- `txindex=1`: Enables a full transaction index for our node, which is useful for querying historical transactions.
+
+After setting up the configuration file, our node will be ready to run in `regtest` mode.
+
```

Comment:
add a comment saying to change to any other network than regtest, set `signet=1` or `testnet=1`. and comment out the `fallbackfee`.